### PR TITLE
✨ Add extraArgs option

### PR DESCRIPTION
**Pull request by zaber-paul**
## Description

This can be used to provide extra arguments to the 'git commit' command, e.g. '--fixup HEAD'

## Tests

Snapshot was updated with `yarn jest --updateSnapshot`
- [x] All tests passed.

### CODE REVIEW

File: src/prompts.js [hunk 1]
```diff
@@ -67,6 +67,11 @@ const gitmoji = (gitmojis) => {
       name: 'reference',
       message: 'Issue / PR reference:',
       validate: (value) => guard.reference(value, configVault.getIssueFormat())
+    },
+    {
+      name: 'extraArgs',
+      message: 'Extra commit args (e.g. --fixup HEAD):',
+      validate: guard.extraArgs
```
Comment (carloscuesta) [inline src/prompts.js]: The `guard.extraArgs` method does not exists in the `guard.js` file, maybe you forgot to add it?

File: test/__snapshots__/gitmojiCli.spec.js.snap [hunk 2]
```diff
@@ -132,6 +132,11 @@ Array [
     "name": "reference",
     "validate": [Function],
   },
+  Object {
+    "message": "Extra commit args (e.g. --fixup HEAD):",
+    "name": "extraArgs",
+    "validate": undefined,
```
Comment (carloscuesta) [inline test/__snapshots__/gitmojiCli.spec.js.snap]: Referring to the comment I posted, that's why this is undefined in the snapshot

**Comment by zaber-paul** [comment]
I removed the validation function since any options (and future git commit options) should be allowed. If you enter incorrect text for `extraArgs`, the commit will fail with a message from `git`. Also, updated the snapshots.

I only ever use the `--fixup` option here (personally), so I'd be OK to rewrite this PR as a dedicated prompt for fixup reference if you prefer (it could have a validation function matching SHAs and branch names). Or, if you prefer to merge I think this PR is in a mergeable state now.

**Comment by carloscuesta** [comment]
Hey @zaber-paul 

Thanks for the work you put into this PR, I really appreciate that❤️ 

I want to focus on making the cli simpler and not introduce a lot of new features.

Thanks!

**Pull request closed without merging**